pkg/jobs/registry.go

```diff
@@ -1454,6 +1454,11 @@ type Resumer interface {
 	// cannot assume that any other methods have been called on this Resumer
 	// object.
 	OnFailOrCancel(ctx context.Context, execCtx interface{}, jobErr error) error
+
+	// CollectProfile is called when a job has been requested to collect a job
+	// profile. This profile may contain any information that provides enhanced
+	// observability into a job's execution.
+	CollectProfile(ctx context.Context, execCtx interface{}) error
```

Comment:
Do we want to have every resumer implement this?
Looks like most of them return nil.
Perhaps we can have 
```
type ProfilerCollector interface {
   CollectProfile(....)
}
```
And have the jobs implement this interface?
When you look at the resumer, check to see if it implemented this method, and call it.  If not, return "nil" as the default implementation already does.

Reply:
I think I prefer the implementation actively, explicitly state that it doesn't want to do anything when profiled, rather than implicitly say so by forgetting to implement the method.